internal/provider/resource_gitlab_user_gpgkey.go

```diff
@@ -0,0 +1,187 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	gitlab "github.com/xanzy/go-gitlab"
+)
+
+var _ = registerResource("gitlab_user_gpgkey", func() *schema.Resource {
+	return &schema.Resource{
+		Description: `The ` + "`" + `gitlab_user_gpgkey` + "`" + ` resource allows to manage the lifecycle of a GPG key assigned to the current user or a specific user.
```

Comment:
I think it makes sense to add a little info box to let readers know that managing keys for arbitrary users requires administrative privileges. WDYT?

```suggestion
		Description: `The ` + "`" + `gitlab_user_gpgkey` + "`" + ` resource allows to manage the lifecycle of a GPG key assigned to the current user or a specific user.
		
-> Managing GPG keys for arbitrary users requires admin privileges.
```